### PR TITLE
Add answers for two common troubleshooting questions

### LINKED ISSUE #57261: Create first version of troubleshooting guide for permissions

### PR DESCRIPTION
**Pull request by pjlast**
Closes #57261

An initial start to having sections for troubleshooting common repository permissions issues. 

## Test plan

Doc updates

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@pjlast/permissions-troubleshooting)

### CODE REVIEW

File: doc/admin/permissions/index.md [hunk 1]
```diff
@@ -157,3 +157,19 @@ the removal of `horsegraph-v1` from the set.
...
+
+The two most common scenarios where users may be able to access repositories on Sourcegraph that they don't have access to on the code host are:
+1. The user is a site admin, and Sourcegraph has not been configured to enforce permissions for site admins. [Configure Sourcegraph to enforce permissions for site admins](#site-administrators).
+2. The code host connection is not configured to enforce repository permissions. [Enable authorization on the code host connection](#getting-started).
+
+For additional insight into why a user can view specific repositories on Sourcegraph, visit their [permissions page](./syncing.md#verify-via-ui) as a site administrator.
+
+### Users cannot access repositories even though they do have access on the code host
+
+In order to sync repository permissions from the code host, Sourcegraph needs to be able to associate each user on Sourcegraph with a user on the code host. To achieve this, an [auth provider needs to be configured](../auth/index.md) for the code host you would like to sync repository permissions for.
```
Comment (eseliger) [inline doc/admin/permissions/index.md:173]:
```suggestion
In order to sync repository permissions from the code host, Sourcegraph needs to be able to associate each user on Sourcegraph with a user on the code host. To achieve this, an [auth provider needs to be configured](../auth/index.md) *for the code host* you would like to sync repository permissions for. Note: A SAML auth provider is NOT a code host provider, and both will be required. 
```

don't love the wording, but maybe you're a better wordsmith than I am

**Review by eseliger — APPROVED** [review]
Thanks for writing these up!

I generally wonder - should we make these very boring step-by-step things with screenshots and so forth? Discussion is not blocking this PR at all, just wondering how much "room for interpretation" we want to leave the consumers.

**Comment by pjlast** [comment]
@eseliger I opted to link to existing docs with the hope that it makes existing docs more discoverable, and so that we don't need to keep the same information updated in multiple places. But thought about separating it out to a dedicated troubleshooting section with screenshots and everything as well, so perhaps there is merit in that

**Comment by mrnugget** [comment]
> I generally wonder - should we make these very boring step-by-step things with screenshots and so forth?

I agree with that. 

I think the troubleshooting docs should be self-contained as a thing that guides the user to the root of a problem. 

Good:

```
Problem: user can't see repository, but has access to it on code host

Steps to troubleshoot:
* Check whether query `repo:my-repo` returns something.
* Check whether repo is cloned.
* Check whether authorization is enabled: ...
* Check whether permissions are up to date for user: ...
```

And each step here can link to other docs too for additional information, or include screenshot, steps.

Bad:

```
Problem: user can't see repository, but has access to it on code host

Check if permissions are synced: [link to generic permissions doc]
```

**Pull request closed without merging**